packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js

```diff
@@ -464,14 +464,20 @@ export default function useAutocomplete(props) {
     },
   );
 
+  const isSameValue = (value1, value2) => {
```

Comment:
Move this method inside the `checkHighlightedOptionExists` method as it's not being used elsewhere currently.